addons/l10n_ro_edi_stock/models/l10n_ro_edi_stock_etransport_interface.py

```diff
@@ -0,0 +1,640 @@
+    l10n_ro_edi_stock_operation_scope = fields.Selection(selection=OPERATION_SCOPES, string="Operation Scope")
+
+    l10n_ro_edi_stock_vehicle_number = fields.Char(string="Vehicle Number", size=20)
+    l10n_ro_edi_stock_trailer_1_number = fields.Char(string="Trailer 1 Number", size=20)
+    l10n_ro_edi_stock_trailer_2_number = fields.Char(string="Trailer 2 Number", size=20)
+
+    l10n_ro_edi_stock_available_start_loc_types = fields.Char(compute='_compute_available_location_types')
+    l10n_ro_edi_stock_start_loc_type = fields.Selection(selection=LOCATION_TYPES, string="Location Type", default='location')
+
+    l10n_ro_edi_stock_available_end_loc_types = fields.Char(compute='_compute_available_location_types')
+    l10n_ro_edi_stock_end_loc_type = fields.Selection(selection=LOCATION_TYPES, string="Location Type", default='location')
+
+    # Data fields for every location type
+    l10n_ro_edi_stock_start_bcp = fields.Selection(selection=BORDER_CROSSING_POINTS, string="Border Crossing Point")
+    l10n_ro_edi_stock_start_customs_office = fields.Selection(selection=CUSTOMS_OFFICES, string="Custom Office")
+    l10n_ro_edi_stock_start_state_id = fields.Many2one(comodel_name='res.country.state', string="State", domain="[('country_id.code', '=', 'RO')]")
+    l10n_ro_edi_stock_start_city = fields.Char(string="City")
+    l10n_ro_edi_stock_start_street = fields.Char(string="Street")
+    l10n_ro_edi_stock_start_zip = fields.Char(string="Postal Code")
+    l10n_ro_edi_stock_start_other_info = fields.Char(string="Other Info")
+
+    l10n_ro_edi_stock_end_bcp = fields.Selection(selection=BORDER_CROSSING_POINTS, string="Border Crossing Point")
+    l10n_ro_edi_stock_end_customs_office = fields.Selection(selection=CUSTOMS_OFFICES, string="Customs Office")
+    l10n_ro_edi_stock_end_state_id = fields.Many2one(comodel_name='res.country.state', string="State", domain="[('country_id.code', '=', 'RO')]")
+    l10n_ro_edi_stock_end_city = fields.Char(string="City")
+    l10n_ro_edi_stock_end_street = fields.Char(string="Street")
+    l10n_ro_edi_stock_end_zip = fields.Char(string="Postal Code")
+    l10n_ro_edi_stock_end_other_info = fields.Char(string="Other Info")
+
+    l10n_ro_edi_stock_remarks = fields.Text(string="Remarks")
+
+    # Compute Methods
+
+    @api.depends('picking_id.move_ids')
+    def _compute_l10n_ro_edi_stock_move_ids(self):
+        for interface in self:
+            interface.l10n_ro_edi_stock_move_ids = interface.picking_id.move_ids
+
+    @api.depends('picking_id.company_id')
+    def _compute_l10n_ro_edi_stock_company_id(self):
+        for interface in self:
+            interface.l10n_ro_edi_stock_company_id = interface.picking_id.company_id
+
+    @api.depends('picking_id.carrier_id')
+    def _compute_l10n_ro_edi_stock_carrier_id(self):
+        for interface in self:
+            interface.l10n_ro_edi_stock_carrier_id = interface.picking_id.carrier_id
```

Comment:
For these fields, instead of having these computes, I think you can just do a `related=...` in the field declaration

Reply:
We can't have these be related fields because in the `l10n_ro_edi_stock_batch` module we need to override these functions and add more logic to them. Which cannot be easily done if these were related fields